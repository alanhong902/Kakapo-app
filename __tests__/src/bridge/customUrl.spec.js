import bridge from 'bridge/web/customUrl';

const { getCustomFile, getCustomURL } = bridge;

test('[bridge/customUrl/getCustomFile]', () => {
  expect(getCustomFile()).toMatchSnapshot();
  expect(getCustomFile(5)).toMatchSnapshot();
});

test('[bridge/customUrl/getCustomURL success]', () => {
  const data = { source: 'file', name: 'Testing' };
  expect(getCustomURL(data)).toMatchSnapshot();
});

test('[bridge/customUrl/getCustomURL failure]', () => {
  const data = { source: 'other', file: 'wind.mp9', name: 'Testing' };
  const err =
    'File is mp9, but must be one of mp3, opus, ogg, wav, aac, m4a, mp4, weba';
  expect(() => getCustomURL(data)).toThrow(err);
});
