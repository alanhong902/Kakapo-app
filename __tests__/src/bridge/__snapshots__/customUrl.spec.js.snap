// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[bridge/customUrl/getCustomFile] 1`] = `undefined`;

exports[`[bridge/customUrl/getCustomFile] 2`] = `undefined`;

exports[`[bridge/customUrl/getCustomURL success] 1`] = `
Object {
  "editing": false,
  "file": null,
  "img": null,
  "link": null,
  "name": "Testing",
  "playing": false,
  "progress": 0,
  "recentlyDownloaded": true,
  "source": "file",
  "tags": null,
  "volume": 0.5,
}
`;
